### PR TITLE
Added Client Pull and fixed uuids

### PR DESCRIPTION
**Pull request by ConnorFoody**
Addresses #4 

Accidentally put a fix for something else in here. Wasn't thinking and used a UUID library that generates 16 character ids.

We should look into compressing our responses with gzip or some binary format.

### CODE REVIEW

File: party/party.go [hunk 1]
```diff
@@ -109,3 +110,33 @@ func (p *Party) SetOwner(userUUID UserUUID) error {
...
+	p.changeID++
+}
+
+// Pull returns the user data in a serializable format.
+// NOTE: this checks for changes before checking uid.
+func (p *Party) Pull(userUUID UserUUID, changeID uint64) (interface{}, error) {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	// TODO: what if change ID is larger?
```
Comment (daniel-centore) [inline party/party.go]: If ID is larger we should error

Comment (daniel-centore) [inline party/party.go]: Or return nil, nil again. The latter is probably easier. It's the client's fault if it's an idiot.

File: server/partyManager.go [hunk 2]
```diff
@@ -56,22 +59,42 @@ func (pm *PartyManager) Party(pid uuid.UUID) (*party.Party, error) {
...
 	delete(pm.parties, pid)
 	return nil
 }
+
+// generateUUID with 6 letters / numbers
+// panics if can't create a uuid
+func (pm PartyManager) generateUUID() PartyUUID {
+	letterBytes := "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	// try to generate a uuid
+	for j := 0; j < 10; j++ {
```
Comment (daniel-centore) [inline server/partyManager.go]: No magic numbers. Make a constant.

File: server/partyManager.go [hunk 3]
```diff
@@ -56,22 +59,42 @@ func (pm *PartyManager) Party(pid uuid.UUID) (*party.Party, error) {
...
+		b := make([]byte, 6)
```
Comment (daniel-centore) [inline server/partyManager.go]: What does 6 mean here?

File: server/server_test.go [hunk 4]
```diff
@@ -63,24 +63,31 @@ func TestServerSingleUser(t *testing.T) {
 
 // TODO: check coverage
 
+// parses the pid
```
Comment (daniel-centore) [inline server/server_test.go]: All your function comments need to be better

Comment (daniel-centore) [inline server/server_test.go]: Well, not all of them. But, like, what is this supposed to do?

File: server/partyManager.go (hunk 3 above)
Comment (ConnorFoody) [inline server/partyManager.go]: Number of characters in the id, I can move this out to a param, or we can make it const

Comment (daniel-centore) [inline server/partyManager.go]: I think const would be best

File: server/server_test.go (hunk 4 above)
Comment (ConnorFoody) [inline server/server_test.go]: ok, will do, got lazy cause its for testing. Are the actual code comments OK?

Comment (daniel-centore) [inline server/server_test.go]: Mostly this file

**Comment by daniel-centore** [comment]
Any reason this isn't merged yet?

**Pull request merged**